lib/crypto/packaged.py

```diff
@@ -51,6 +52,17 @@ def get_endpoint(server):
     return u'{server}/1.0/sign_addon'.format(server=server)
 
 
+def get_id(addon):
+    """Return the addon GUID if <= 64 chars, or its sha256 hash otherwise.
+
+    We don't want GUIDs longer than 64 chars: bug 1203365.
+    """
+    guid = addon.guid
+    if len(guid) <= 64:
+        return guid
+    return hashlib.sha256(guid).hexdigest()
```

Comment:
The client decides what the CN should be based on the ID the same way that
we do, so it doesn't need any other data.
On Sep 10, 2015 16:42, "Andy McKay" <notifications@github.com> wrote:

> In lib/crypto/packaged.py
> <https://github.com/mozilla/olympia/pull/727#discussion_r39228086>:
>
> > @@ -51,6 +52,17 @@ def get_endpoint(server):
> >      return u'{server}/1.0/sign_addon'.format(server=server)
> >
> >
> > +def get_id(addon):
> > +    """Return the addon GUID if <= 64 chars, or its sha256 hash otherwise.
> > +
> > +    We don't want GUIDs longer than 64 chars: bug 1203365.
> > +    """
> > +    guid = addon.guid
> > +    if len(guid) <= 64:
> > +        return guid
> > +    return hashlib.sha256(guid).hexdigest()
>
> Does the id need to be prefixed with sha256: or something so that the
> client can detect if it has been hashed? Or will it be able to do that by
> looking at other data and know this has been hashed?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/olympia/pull/727/files#r39228086>.
>